biolink/pieces-parts/rna-seq-test.rkt

```diff
@@ -168,232 +219,93 @@ purpose: to find any regulatory relationships where genes on a given list regula
-        (define direct-interaction? (set-member? direct-interaction-preds (edge->pred  e)))
-        ;;record provenance
-        (define direct-interaction-provenance (pubmed-ids-from-edge e))
-        ;;add information to the hash
-        (hash-update! gene-regulation-scores
-                      (assoc g (curie-synonyms/names g))
-                      (lambda (v) (set-add v (list (assoc rg (curie-synonyms/names rg))
-                                                   (list 0 direction-provenance)
-                                                   (list direct-interaction? direct-interaction-provenance)
-                                                   (concept->curie (edge->subject e))))) '())
-        )
-      (edges/query q 'G->rG-on-list)
-      )
-
+     (1-hop-edges->hash (edges/query q 'G+-pos->rG-on-list) 1)
+     (1-hop-edges->hash (edges/query q 'G--neg->rG-on-list) -1)
+     (1-hop-edges->hash (edges/query q 'G->rG-on-list) 0)
      )
-   genes
+   uniprot-list
    )
   ;;sort list by the number of regulated genes a gene influences (most to least)
-  (define sorted-gene-regulation-scores (sort
-                                         (hash-map gene-regulation-scores (lambda (k v) (cons k v)))
-                                         >
-                                         #:key (lambda (x) (length (set-intersect (map caar (cdr x)) genes)))))
-  
-  sorted-gene-regulation-scores
+  (sort
+   (hash-map gene-regulation-scores (lambda (k v) (cons k v)))
+   >
+   #:key (lambda (x) (length (cdr x))))
   )
-#|Examples
-(define 1-hop-mini-test (time (find-1-hop-relations (take uniprots 10))))
-(define 1-hop-full-test (time (find-1-hop-relations uniprots)))
-|#
-
 #|
-Finds all cell/tissue types (UBERON) that each gene is expressed in
-|#
-(define (expression-locations genes-and-proteins)
-  (define expression-location (make-hash))
-  (for-each
-   (lambda (g)
-     (define q (time (query/graph
-                      ((G-and-rG g)
-                       (C #f))
-                      ((G-and-rG->C '("expressed_in")))
-                      (G-and-rG G-and-rG->C C))))
-     (for-each
-      (lambda (e)
-        (unless (not (string-prefix? (concept->curie (edge->object e)) "UBERON:"))
-          (define g (concept->curie (edge->subject e)))
-          (define t (concept->curie (edge->object e)))
-          (hash-update! expression-location (assoc g (curie-synonyms/names g)) (lambda (v) (set-add v (assoc t (curie-synonyms/names t)))) '())
-          )
-        )
-      (edges/query q 'G-and-rG->C)
-      )
-     )
-   genes-and-proteins
-   )
-  expression-location
-  )
-#|Examples
-(define cell/tissue-types-mini-test (expression-locations (take uniprots 20)))
-(define cell/tissue-types-full-test (expression-locations uniprots))
+(define 1-hop-mini-test (time (1-hop-query/report (take uniprots 100))))
+(define 1-hop-results (time (1-hop-query/report uniprots)))
 |#
 
+#|
```

Comment:
Why is this commented out?

Reply:
I haven't finished testing find-intersections, but I'll send another pull request once I finish that function